pkg/statistics/handle/usage/predicatecolumn/predicate_column.go

```diff
@@ -0,0 +1,184 @@
+	return loadColumnStatsUsage(sctx, loc, query)
+}
+
+// LoadColumnStatsUsageForTable loads column stats usage information for a specific table from disk.
+func LoadColumnStatsUsageForTable(sctx sessionctx.Context, loc *time.Location, tableID int64) (map[model.TableItemID]statstypes.ColStatsTimeInfo, error) {
+	query := "SELECT table_id, column_id, CONVERT_TZ(last_used_at, @@TIME_ZONE, '+00:00'), CONVERT_TZ(last_analyzed_at, @@TIME_ZONE, '+00:00') FROM mysql.column_stats_usage WHERE table_id = %?"
+	return loadColumnStatsUsage(sctx, loc, query, tableID)
+}
+
+// GetPredicateColumns returns IDs of predicate columns, which are the columns whose stats are used(needed) when generating query plans.
+func GetPredicateColumns(sctx sessionctx.Context, tableID int64) ([]int64, error) {
+	// Each time we retrieve the predicate columns, we also attempt to remove any column stats usage information whose column is dropped.
+	err := cleanupDroppedColumnStatsUsage(sctx, tableID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	rows, _, err := utilstats.ExecRows(
+		sctx,
+		"SELECT column_id, CONVERT_TZ(last_used_at, @@TIME_ZONE, '+00:00') FROM mysql.column_stats_usage WHERE table_id = %? AND last_used_at IS NOT NULL",
+		tableID,
+	)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	columnIDs := make([]int64, 0, len(rows))
+	for _, row := range rows {
+		// Usually, it should not be NULL.
+		// This only happens when the last_used_at is not a valid time.
+		if row.IsNull(1) {
+			continue
+		}
+		colID := row.GetInt64(0)
+		columnIDs = append(columnIDs, colID)
+	}
+	return columnIDs, nil
+}
+
+// cleanupDroppedColumnStatsUsage deletes the column stats usage information whose column is dropped.
+func cleanupDroppedColumnStatsUsage(sctx sessionctx.Context, tableID int64) error {
+	is := sctx.GetDomainInfoSchema().(infoschema.InfoSchema)
+	table, ok := is.TableByID(tableID)
+	if !ok {
+		// Usually, it should not happen.
+		// But if it happens, we can safely do nothing.
+		return nil
+	}
+	allColumns := table.Meta().Columns
+	// Due to SQL limitations, column IDs must be converted to strings for proper escaping in the query :(
+	columnIDs := make([]string, 0, len(allColumns))
+	for _, col := range allColumns {
+		columnIDs = append(columnIDs, fmt.Sprintf("%d", col.ID))
+	}
+
+	// Delete the column stats usage information whose column is dropped.
+	_, _, err := utilstats.ExecRows(
+		sctx,
+		"DELETE FROM mysql.column_stats_usage WHERE table_id = %? AND column_id NOT IN (%?)",
+		tableID,
+		columnIDs,
+	)
+
+	return err
+}
+
+// SaveColumnStatsUsageForTable saves column stats usage information for a specific table to disk.
+func SaveColumnStatsUsageForTable(
+	sctx sessionctx.Context,
+	colStatsUsage map[model.TableItemID]statstypes.ColStatsTimeInfo,
+) error {
+	for colID, statsUsage := range colStatsUsage {
+		lastUsedAt := "NULL"
```

Comment:
I am not sure if we have a better way to handle this. The ExecRows function cannot handle nil correctly, so I have to rely on the CONVERT_TZ to insert 'null' if it is 'NULL'.